### PR TITLE
Npmify for dictionary viewer

### PR DESCRIPTION
**Pull request by cheapsteak**
- Using npm scripts instead of mkdocs commands for build/dev/publish
- Updated readme with instructions for npm commands
- webpack config
- added package.json
- uglify js that are built by webpack

new folders:

```
/src
  /js
    # new javascript source files should go here, can require from `node_modules`
  /styles
     # stylus files go here, they can include/import files from `node_modules`
```

To add more entry points: 

https://github.com/icgc-dcc/dcc-docs/blob/a1a950dda296c2bc48433e6671fa90c9c1f84039/config/webpack.js#L19-L21

Usage example: 
https://github.com/icgc-dcc/dcc-docs/blob/a1a950dda296c2bc48433e6671fa90c9c1f84039/src/styles/dictionary-viewer.styl#L1
https://github.com/icgc-dcc/dcc-docs/blob/a1a950dda296c2bc48433e6671fa90c9c1f84039/src/js/dictionary-viewer.js#L1

### CODE REVIEW

**Comment by btiernay** [comment]
Love the term "npmify"

File: config/webpack.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+var PROD = process.env.NODE_ENV === 'production';
+
+var plugins = [
+  new webpack.optimize.CommonsChunkPlugin("commons-chunk.js"),
+
+  // Can't uglify yet, dictionary viewer needs to use ngannotate
+  // PROD && new webpack.optimize.UglifyJsPlugin({
+  //   beautify: false,
+  //   mangle: { screw_ie8 : false },
```
Comment (btiernay) [inline config/webpack.js:11]: Is `screw_ie8` PC?!?!

File: package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "url": "git+https://github.com/icgc-dcc/dcc-docs.git"
+  },
+  "author": "",
+  "license": "UNLICENSED",
+  "bugs": {
+    "url": "https://github.com/icgc-dcc/dcc-docs/issues"
+  },
+  "homepage": "https://github.com/icgc-dcc/dcc-docs#readme",
+  "devDependencies": {
+    "@icgc/dictionary-viewer": "1.0.3",
```
Comment (btiernay) [inline package.json:35]: Curious, what does the `@` mean?

File: config/webpack.js (hunk 1 above)
Comment (cheapsteak) [inline config/webpack.js:11]: lol, it's uglify's choice :P

File: package.json (hunk 2 above)
Comment (cheapsteak) [inline package.json:35]: @xxxxx/yyyyyy denotes a scoped package named yyyyy belonging to the user/organization xxxxx

Scoped packages allow us to not pollute the global npm namespace with whatever names we choose for our icgc-specific stuff, and also allow us to name our packages whatever without having to worry about whether the name has already been taken :)

**Pull request merged**